pkg/apis/infra/v1alpha1/cluster.go

```diff
@@ -47,29 +45,30 @@ type CustomClusterSpec struct {
 	// TODO: any other configurations that does not exist in upstream cluster api
 
 	// MachineRef is the reference of nodes for provisioning a kurator cluster.
-	// +optional
 	MachineRef *corev1.ObjectReference `json:"machineRef,omitempty"`
 
 	// CNIConfig is the configuration for the CNI of the cluster.
 	CNI CNIConfig `json:"cni"`
 
 	// ControlPlaneConfig contains control plane configuration.
+	// +optional
 	ControlPlaneConfig ControlPlaneConfig `json:"controlPlaneConfig,omitempty"`
```

Comment:
if this is optional, should it be a pointer? cc @hzxuzhonghu

Reply:
Yes, we can do that